### DISCUSSION TITLE
Data Collection (if any) from MQTTX Web

### DISCUSSION THREAD

**Discussion by jason-grimme** [emqx/MQTTX#1674] [General]
Some people may be concerned about using the web interface with their message broker if they are unsure how the input/output is used. 

It would be nice if the public web interface (http://mqtt-client.emqx.com/) had some disclaimer describing the data collection policy, if any.  If no data is collected, it would be nice if that was stated.  Additionally, it would be nice if there was a message like "All data is processed on client side, it never leaves your browser" (assuming this is true).  Similar messages exist for jwt.io & jwt.ms .  Obviously people who are concerned can run the GUI/CLI/Web locally, but if the data policy was clearly defined then many would not feel the need to. 

 
While I have faith that credentials are not collected, it would be reasonable for MQTTX team to want to know in general how utilized the web utility is used.    For me, collecting general usage would be reasonable, but input/output (credential, key, topic, message, etc) would not.

**Comment by ysfscream** [comment]
Hi @jason-grimme,

Thank you for your feedback. In future updates, we will add a disclaimer in both the UI and the documentation. For MQTTX Web, all data remains local to your machine; we do not perform cloud operations or data processing. Any message data is solely related to the service you are connected to. Additionally, while we do use GTM and have GA configured, these only collect page visit metrics and do not involve any user privacy data. Thank you for your trust. 💛

Best regards

**Reply by jason-grimme** [reply]
Thank you, the current practice seems very reasonable to me.  Stating this clearly will be a great addition to MQTTX Web.
